src/progress.rs

```diff
@@ -841,4 +1149,101 @@ mod test_progress_set {
         assert_eq!(pre, *set.get(1).expect("Peer should not have been deleted"));
         Ok(())
     }
+
+    #[test]
+    fn test_membership_change_configuration_remove_voter() -> Result<()> {
+        check_membership_change_configuration((vec![1, 2], vec![]), (vec![1], vec![]))
+    }
+
+    #[test]
+    fn test_membership_change_configuration_remove_learner() -> Result<()> {
+        check_membership_change_configuration((vec![1], vec![2]), (vec![1], vec![]))
+    }
+
+    #[test]
+    fn test_membership_change_configuration_conflicting_sets() {
+        assert!(
+            check_membership_change_configuration((vec![1], vec![]), (vec![1], vec![1]),).is_err()
+        )
+    }
+
+    #[test]
+    fn test_membership_change_configuration_empty_sets() {
+        assert!(check_membership_change_configuration((vec![], vec![]), (vec![], vec![])).is_err())
+    }
+
+    #[test]
+    fn test_membership_change_configuration_empty_voters() {
+        assert!(
+            check_membership_change_configuration((vec![1], vec![]), (vec![], vec![]),).is_err()
+        )
+    }
```

Comment:
The above two error cases are testing for errors in the starting configuration, but there are not similar tests for the ending configuration, and it looks to me the starting and ending cases have different code paths. If so, can you add more of these same tests but for the ending set?